File: Python/34. Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        nums_sorted = sorted(nums)
        first, last = -1, -1

        for i in range(len(nums_sorted)):
            if target == nums_sorted[i] and (i == 0 or (i > 0 and nums_sorted[i - 1] != target)):
                first = i
            if target == nums_sorted[i] and (i == len(nums_sorted) - 1 or (i < len(nums_sorted) - 1 and nums_sorted[i + 1] != target)):
                last = i

        return [first, last]

File: C++/34. Find First and Last Position of Element in Sorted Array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target)
    {
        sort(nums.begin(), nums.end());
        int first = -1, last = -1;
        for (int i = 0;i < nums.size();++i)
        {
            if (target == nums[i] && (i == 0 || (i > 0 && nums[i - 1] != target)))
                first = i;
            if (target == nums[i] && (i == nums.size() - 1 || (i < nums.size() - 1 && nums[i + 1] != target)))
                last = i;
        }
        return vector<int>({ first,last });
    }
};